### PR TITLE
feat(core): management user verifications

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add user verification manage API:

1. `GET /users/:id/mfa-verifications` returns `mfaVerifications` without sensitive informations
2. `DELETE /users/:id/mfa-verifications/:verificationId`

Notice that integration tests are unavailable now, because it is blocked by the verification binding flow.

## Testing

Local tested and CI tests.

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] integration tests~~ (see description above)
~~- [ ] docs~~
- [x] unit tests

### CODE REVIEW

File: packages/core/src/routes/admin-user.ts [hunk 1]
```diff
@@ -42,7 +43,10 @@ export default function adminUserRoutes<T extends AuthedRouter>(
 
       const user = await findUserById(userId);
 
-      ctx.body = pick(user, ...userInfoSelectFields);
+      ctx.body = {
+        ...pick(user, ...userInfoSelectFields),
+        verifications: transpileUserProfileVerifications(user),
```
Comment (simeng-li) [inline packages/core/src/routes/admin-user.ts:48]: Maybe `mfaVarifications`?  Would verifications be confused with email/SMS passcode verifications?

Comment (simeng-li) [reply packages/core/src/routes/admin-user.ts:48]: Should we create a `getUserProfile` library method or a util method for this? 

We have multiple endpoints returning `userProfileResponse`,  not just get user by id.

File: packages/schemas/src/types/user.ts [hunk 2]
```diff
@@ -25,6 +26,17 @@ export type UserInfo = z.infer<typeof userInfoGuard>;
...
+    .array(
+      // Zod discriminatedUnion is too complicated for swagger
+      z.object({
+        ...baseVerification,
+        type: z.nativeEnum(MfaFactor),
+        agent: z.string().optional(),
+        used: z.boolean().optional(),
+      })
+    )
+    .optional(),
```
Comment (simeng-li) [inline packages/schemas/src/types/user.ts:39]: should not be optional?

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 3]
```diff
@@ -186,12 +186,12 @@ export const identitiesGuard = z.record(identityGuard);
...
 export type Identities = z.infer<typeof identitiesGuard>;
 
-const baseVerification = {
+export const baseVerification = {
   id: z.string(),
   createdAt: z.date(),
 };
 
-const verificationGuard = z.discriminatedUnion('type', [
+export const verificationGuard = z.discriminatedUnion('type', [
```
Comment (simeng-li) [inline packages/schemas/src/foundations/jsonb-types.ts:194]: Better give a more specific name.  

`import {verificationGuard, baseVerification} from @logto/schemas` 

Hard to tell what verification stands for without any context.

File: packages/core/src/routes/admin-user.ts [hunk 4]
```diff
@@ -301,4 +305,36 @@ export default function adminUserRoutes<T extends AuthedRouter>(
       return next();
     }
   );
+
+  router.delete(
+    '/users/:userId/mfaVerifications/:verificationId',
```
Comment (xiaoyijun) [inline packages/core/src/routes/admin-user.ts:310]:
```suggestion
    '/users/:userId/mfa-verifications/:verificationId',
```

File: packages/core/src/routes/admin-user.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts]: Updated to `mfaVerifications`.

Currently, only this endpoint need this field.

File: packages/schemas/src/types/user.ts (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/src/types/user.ts:39]: Currently, only some endpoints need this field.

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 3 above)
Comment (wangsijie) [reply packages/schemas/src/foundations/jsonb-types.ts]: updated

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/schemas/src/types/user.ts (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/src/types/user.ts]: updated with new endpoint

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 5]
```diff
@@ -186,12 +186,12 @@ export const identitiesGuard = z.record(identityGuard);
 export type Identity = z.infer<typeof identityGuard>;
 export type Identities = z.infer<typeof identitiesGuard>;
 
-const baseMfaVerification = {
+export const baseMfaVerification = {
   id: z.string(),
-  createdAt: z.date(),
+  createdAt: z.string(),
```
Comment (charIeszhao) [inline packages/schemas/src/foundations/jsonb-types.ts:191]: Why change it to string?

**Review by simeng-li — APPROVED** [review]

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**